src/components/Auth/Auth.js

```diff
@@ -0,0 +1,95 @@
+import { h, Component } from 'preact'
+import { AuthCheckProcessor } from './AuthCheckProcessor'
+import { FaceTecSDK } from '../../../core-sdk/FaceTecSDK.js/FaceTecSDK'
+import { Config } from './AuthConfig'
+import { FaceTecStrings } from './assets/FaceTecStrings'
+import { Button } from '@onfido/castor-react'
+
+export default class AuthCapture extends Component {
+  state = {
+    authConfig: {},
+    sessionInit: false,
+    sessionState: 'Initializing...',
+  }
+  componentDidUpdate(prevProps, prevState) {
+    if (
+      Object.keys(prevState.authConfig).length === 0 &&
+      Object.keys(this.state.authConfig).length > 0 &&
+      FaceTecSDK.getStatus() === 0
+    ) {
+      this.initFaceTec()
+    }
+  }
```

Comment:
I agree with @ethan605'c comment plus I would use hooks

Reply:
Please refer to [this PR](https://github.com/onfido/onfido-sdk-ui/pull/1431/commits/44cae6abcdb83da897695e7e89ef1912be55fdca) to take a look at this implementation :) cheers ⭐